src/app/components/header/header.component.ts

```diff
@@ -10,9 +11,21 @@ import {PopupNotificationService} from '../../services/popup-notification.servic
 export class HeaderComponent {
   @ViewChild('header') header;
   private headerPosition = 0;
-  private contentType = HeaderContentTypes.EDIT;
+  private post: Post = new Post();
 
-  constructor() {}
+  constructor(private postApi: PostApi,
+              private route: Router) {}
+
+  addPost() {
+    this.postApi.createPost(this.post).subscribe(
```

Comment:
This code have a complex logic. Need to migrate it to PostService and call PostApi from this one.